aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -1223,6 +1223,64 @@ void linalg_eigh_cusolver(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& inf
   }
 }
 
+// The 'apply_' word is used for templated by dtype functions that call an API routine
+// underneath. Since the cusolver API has a slightly different structure we do not prepend
+// apply_ to this function.
+void lu_cusolver_looped(const Tensor& self, const Tensor& pivots, const Tensor& infos, bool get_pivots) {
+  AT_DISPATCH_FLOATING_TYPES(
+    self.scalar_type(),
+    "lu_cusolver",
+    [&self,
+     &pivots,
+     &infos,
+     &get_pivots]() {
+    int m = cuda_int_cast(self.size(-2), "m");
+    int n = cuda_int_cast(self.size(-1), "n");
+    int lda = std::max<int>(1, m);
+    int64_t self_stride = matrixStride(self);
+    int64_t batch_size = batchCount(self);
+    scalar_t* self_data = self.data_ptr<scalar_t>();
+    int* infos_data = infos.data_ptr<int>();
+
+    for (auto batch = decltype(batch_size){0}; batch < batch_size; ++batch) {
+      auto handle = at::cuda::getCurrentCUDASolverDnHandle();
```

Comment:
Xiao recommended putting the handle outside the loop in the other PR until we verify and implement proper CUDA stream-based parallelism for batches.